src/commit-body.ts

```diff
@@ -1,21 +1,80 @@
+import Ajv, {JTDParser, JTDSchemaType} from 'ajv/dist/jtd'
+import {InvalidCommitBodyError} from './errors'
 import {Message} from './message'
 
+interface CommitBodyMetaData {
+  job_number?: number
+  job_commit?: string
+}
+
+interface CommitBodyData {
+  namespace: 'git-queue.commit-body'
+  version: number
+  metadata: CommitBodyMetaData
+  payload: string
+}
+
+const CommitBodySchema: JTDSchemaType<CommitBodyData> = {
+  properties: {
+    namespace: {enum: ['git-queue.commit-body']},
+    version: {type: 'int32'},
+    payload: {type: 'string'},
+    metadata: {
+      optionalProperties: {
+        job_number: {type: 'int32'},
+        job_commit: {type: 'string'}
+      },
+      additionalProperties: true
+    }
+  },
+  additionalProperties: true
+}
+
 export class CommitBody {
-  text: string
+  body: CommitBodyData
+  private static parse: JTDParser<CommitBodyData> = new Ajv().compileParser(
+    CommitBodySchema
+  )
 
   constructor(text: string) {
-    this.text = text
+    this.body = this.getParsedBodyData(text)
   }
 
   static fromMessage(message: Message): CommitBody {
-    return new CommitBody(message.getPayload())
+    return new CommitBody(
+      JSON.stringify({
+        namespace: 'git-queue.commit-body',
+        version: 1,
+        metadata: {},
+        payload: message.getPayload()
+      })
+    )
+  }
+
+  getParsedBodyData(text): CommitBodyData {
+    const parsedBody = CommitBody.parse(text)
+    if (this.isCommitBodyData(parsedBody)) {
+      return parsedBody
+    } else {
+      throw new InvalidCommitBodyError(text)
+    }
+  }
+
+  isCommitBodyData(
+    object: CommitBodyData | undefined
+  ): object is CommitBodyData {
+    return typeof object != 'undefined'
   }
 
   toString(): string {
-    return this.text
+    return JSON.stringify(this.body)
+  }
+
+  getPayload(): string {
+    return this.body?.payload
   }
 
   equalsTo(other: CommitBody): boolean {
-    return this.text === other.text
+    return JSON.stringify(this.body) === JSON.stringify(other.body)
```

Comment:
@yeraydavidrodriguez I thought there would be a better way to compare two objects with the same interface in TypeScript. The only problem I see is if we want more flexibility, but I think the strictest comparison is OK here.

Reply:
I will look into it to use a nicer way

Reply:
This was solved using the [isEqual comparison method of Lodash](https://lodash.com/docs/4.17.15#isEqual).